src/Server/ServerType.cpp

```diff
@@ -64,28 +61,50 @@ bool ServerType::shouldStart(Type server_type, const std::string & server_custom
             default:
                 return false;
         }
-    }
+    };
 
-    if (type == Type::QUERIES_CUSTOM)
+    auto is_type_custom = [](Type current_type)
     {
-        switch (server_type)
-        {
-            case Type::CUSTOM:
-                return true;
-            default:
-                return false;
-        }
+        return current_type == Type::CUSTOM;
+    };
+
+    if (exclude_types.contains(Type::QUERIES_ALL))
+        return false;
+
+    if (exclude_types.contains(Type::QUERIES_DEFAULT) && is_type_default(server_type))
+        return false;
+
+    if (exclude_types.contains(Type::QUERIES_CUSTOM) && is_type_custom(server_type))
+        return false;
```

Comment:
Won't we refuse all custom server types here?

E.g. for `... EXCEPT CUSTOM 'aaa'` we put `Custom` into `exclude_types` and when checking `shouldStart` for `CUSTOM 'bbb'` we will return `false` here.

But seems last added test case `START LISTEN QUERIES DEFAULT EXCEPT` check it, does it?

Reply:
`QUERIES CUSTOM` is `Type::QUERIES_CUSTOM` and `CUSTOM 'abc'` is `Type::CUSTOM`.

So if we put the first one we will definitely ignore all custom types (return false), and if we put the second one we will only return false if the name is the same later in the code